core/metadata.go

```diff
@@ -41,6 +41,12 @@ func (meta Metadata) Delete(key string) {
 	delete(meta, key)
 }
 
+// HasValue returns true if the given key exists
+func (meta Metadata) HasValue(key string) bool {
```

Comment:
FWIW, I'm more bothered by the fact that ` func (meta Metadata) GetValue(key string) []byte` doesn't distinguish between empty and inexistent values. But changing its behavior now might of course break something.